feat: 게시글 작성 기능 구현

dhfkdlsj opened the pull request:
## 개요
- 게시글 작성 기능 구현

Resolves: #2(게시글 작성 기능 구현)

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

dhfkdlsj committed:
Feat: 공통 엔티티 생성 시간/수정 시간 자동 관리 기능 추가

- JPA Auditing 기능을 활용해 BaseEntity 구성
- 생성 시 createdAt 자동 저장
- 수정 시 updatedAt 자동 업데이트
- 일부 엔티티는 updatedAt 필드가 필요 없어 선택적으로 상속 가능하도록 설계

Resolves: #2

dhfkdlsj committed:
Feat: Post 엔티티 생성

- 게시글 정보를 저장하는 Post 엔티티 생성
- 카테고리는 Enum 타입으로 관리

Resolves: #2

dhfkdlsj committed:
Style: 개행 추가

Resolves: #2

dhfkdlsj committed:
Feat: @EnableJpaAuditing 추가

- Spring Data JPA의 Auditing 기능을 활성화하기 위해
  @EnableJpaAuditing 어노테이션을 추가

Resolves: #2

dhfkdlsj committed:
Feat: 게시글 생성 기능 구현

- 게시글 생성 기능 구현
- 멤버 부분은 아직 주석 처리

Resolves: #2

dhfkdlsj committed:
Fix: merge conflict 해결

- git pull 중 발생한 충돌 파일 정리
- 변경사항 병합 및 정상 동작 확인

Resolves: #2

dhfkdlsj committed:
Chore: PostReqDto 이름 변경

- PostReq 에서 PostReqDto로 변경

Resolves: #2

dhfkdlsj committed:
Feat: MapStruct 기반 PostReqDto → Post 매핑 기능 도입

- MapStruct 의존성 추가
- PostMapper 인터페이스 생성 및 viewCount, likeCount, deleted 필드 기본값 설정
- PostService에서 Builder 패턴 제거 후 Mapper 적용

Resolves: #2

dhfkdlsj committed:
Merge branch 'dev' of https://github.com/itcen-7th/Cen-Square into feat/#2/post

dhfkdlsj committed:
Feat: 인증된 사용자 정보 기반 게시글 등록 기능 적용

- @AuthenticationPrincipal 어노테이션을 활용하여 CustomUserDetails로부터 회원 정보 추출
- PostReqDto 내에서 memberId를 제거하고, 로그인된 사용자 정보를 통해 서비스 로직에서 처리
- 보안 강화를 위해 요청 바디에 사용자 정보를 포함하지 않도록 수정

Resolves: #2

dhfkdlsj committed:
Feat: 게시글 생성 시 작성자 ID 반환 기능 추가

- 게시글 작성 시 작성자의 식별자(Member ID)를 응답에 포함되도록 로직 수정

Resolves: #2

dhfkdlsj committed:
Feat: 게시글 생성 요청 DTO에 입력값 검증 어노테이션 추가

- title, content, category 필드에 @NotBlank, @NotNull 등 검증 어노테이션 적용
- 유효하지 않은 요청에 대해 400 Bad Request 응답 유도
- 서비스 레이어 도달 전 입력값 필터링으로 안정성 향상

Resolves: #2

dhfkdlsj committed:
Refactor: 중복 BaseEntity 제거

- 중복 BaseEntity 제거

Resolves: #2

heodongwon reviewed (approved):
고생하셨습니다

hj-k66 commented on src/main/java/com/itcen/censquare/domain/post/dto/PostReqDto.java:
```diff
@@ -0,0 +1,23 @@
+  @NotNull(message = "카테고리를 선택해주세요.")
+  private Category category;
+
+  @NotBlank(message = "제목을 입력해주세요.")
+  @Size(max = 255, message = "제목은 255자 이내여야 합니다.")
+  private String title;
+
```
혹시 제목 100자이내면 어떨까요?? 제목이 GPT input으로 들어가서 넘 길면 토큰 많이 쓸 것 같아서용

dhfkdlsj replied:
말씀해주신 부분 저도 공감합니다.
회의 시간에 다 함께 논의해보면 좋을 것 같아요 :)

dhfkdlsj committed:
Refactor: 게시글 제목 길이 제한을 255자 → 54자로 변경

- Post 엔티티의 title 필드 길이 제한을 기존 255자에서 54자로 조정
- GPT input 처리 효율 및 토큰 사용 최적화를 고려한 변경

Resolves: #2
